### PR TITLE
[meshoptimizer] Add basisu support to gltfpack

### PR DESCRIPTION
**Pull request by Honeybunch**
**Describe the pull request**
When using the gltfpack feature of the meshoptimizer port the gltfpack executable will be built with support for basis universal supercompression. This also requires updating the meshoptimizer port to the latest version.

- #### What does your PR fix?
  gltfpack not being able to compress textures

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by Honeybunch** [comment]
I found an issue with this. Please don't merge yet :)

**Comment by Honeybunch** [comment]
Issue I ran into is detailed in a bug to upstream here: https://github.com/zeux/meshoptimizer/issues/507

**Comment by Honeybunch** [comment]
This PR should be good to go. I made sure to test against some test assets and the issue I ran into is no longer a problem.

**Comment by JonLiu1993** [comment]
@Honeybunch, When I tested the features by command "./vcpkg install meshoptimizer[*]" , i got the error log,
please take a look:

````
F:\Feature-test\meshoptimizer\vcpkg\buildtrees\meshoptimizer\src\2309e13dd2-9c815cabbc.clean\encoder/basisu_enc.cpp(1930): error C2220: the following warning is treated as an error
F:\Feature-test\meshoptimizer\vcpkg\buildtrees\meshoptimizer\src\2309e13dd2-9c815cabbc.clean\encoder/basisu_enc.cpp(1930): warning C4389: '==': signed/unsigned mismatch
F:\Feature-test\meshoptimizer\vcpkg\buildtrees\meshoptimizer\src\2309e13dd2-9c815cabbc.clean\encoder/pvpngreader.cpp(166): warning C4244: 'argument': conversion from 'uint64_t' to 'const unsigned int', possible loss of data
F:\Feature-test\meshoptimizer\vcpkg\buildtrees\meshoptimizer\src\2309e13dd2-9c815cabbc.clean\encoder/pvpngreader.cpp(181): warning C4244: 'argument': conversion from 'uint64_t' to 'size_t', possible loss of data
F:\Feature-test\meshoptimizer\vcpkg\buildtrees\meshoptimizer\src\2309e13dd2-9c815cabbc.clean\encoder/pvpngreader.cpp(185): warning C4244: 'return': conversion from 'uint64_t' to 'size_t', possible loss of data
F:\Feature-test\meshoptimizer\vcpkg\buildtrees\meshoptimizer\src\2309e13dd2-9c815cabbc.clean\encoder/pvpngreader.cpp(252): warning C4244: 'argument': conversion from 'uint64_t' to 'size_t', possible loss of data
F:\Feature-test\meshoptimizer\vcpkg\buildtrees\meshoptimizer\src\2309e13dd2-9c815cabbc.clean\encoder/pvpngreader.cpp(256): warning C4244: 'return': conversion from 'uint64_t' to 'size_t', possible loss of data
ninja: build stopped: subcommand failed.
````

**Comment by Honeybunch** [comment]
Very strange! Do you have any more details? What's the default triplet you're using? What compiler is being used? I just used `vcpkg install meshoptimizer[*]:x64-windows` and it succeeded with no errors. I have a fairly recent VS2022 install as well as a VS2019 install. I tried in my default `cmder` environment as well as the VS2022 developer command prompt without issue, making sure to clear my build archive.

**Comment by Honeybunch** [comment]
Aw shoot I see what's happening. `/WX` is an optional parameter set when `MESHOPT_WERROR` is set *but only in upstream*. The revision that's pulled for this predates that parameter and `/WX` is always defined. This is annoying because latest from upstream was causing some runtime failures. Let me see if I can find a happy balance here

**Comment by Honeybunch** [comment]
Yeah if I bump this up to the change that solves this problem I run into runtime errors described in https://github.com/zeux/meshoptimizer/issues/507 the better solution may just be a quick patch to remove `/WX` from the v0.18 version.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Comment by Honeybunch** [comment]
I see that the author-response label was added here but I'm not sure what feedback there is to respond to?

File: ports/meshoptimizer/portfile.cmake [hunk 1]
```diff
@@ -4,8 +4,24 @@ vcpkg_from_github(
...
+# meshopt lib it shouldn't be too nasty to just grab this repo
+if ("gltfpack" IN_LIST FEATURES)
+  vcpkg_from_github(
+      OUT_SOURCE_PATH BASISU_PATH
+      REPO zeux/basis_universal
+      REF 91ca86492bc046bf1d096067a1adcd2309e13dd2
+      SHA512 fe80533db60b40cdc72a64f766c2447ce5c923d84467a926c2e8af4ec42e278d9fa9823b41b3fc7d9b740dd2d41d2f606f0f9990f94d2398f253bc86350a4287
+      HEAD_REF gltfpack
+  )
+endif()
```
Comment (JavierMatosD) [inline ports/meshoptimizer/portfile.cmake:23]: Why can't we use our copy of basisu?

Comment (Honeybunch) [reply ports/meshoptimizer/portfile.cmake:23]: Gltfpack requires a specific branch of this fork of basisu to function. On top of that basis_universal does not have an upstream cmake integration that's setup to be installed properly and the current port is out of date and heavily modified. I have a PR upstream there to fix up the cmake but it hasn't been touched.

**Pull request merged**